app/src/main/java/com/pvp/app/service/ExerciseServiceImpl.kt

```diff
@@ -28,4 +53,14 @@ class ExerciseServiceImpl @Inject constructor(
             )
         )
     }
+
+    override suspend fun getInfrequentActivities(maxOccurrence: Int): List<SportActivity> {
+        val activityOccurrences = getActivitiesWithOccurrencesMap()
+
+        return SportActivity.entries.filter { activity ->
+            (activityOccurrences[activity] == null
+                    || activityOccurrences[activity]!! <= maxOccurrence)
+                    && activity.id != 0
```

Comment:
What does `0` id stand for? It most likely would be better to reference exact SportActivity and then get its id to not hard-code it and make it more readable.

`SportActivity.X.id`

Reply:
It stands for `Other`, helps handle activities we do not support.